helix-core/src/main/java/org/apache/helix/HelixTimerTask.java

```diff
@@ -32,4 +32,9 @@
    * Stop a timer task
    */
   public abstract void stop();
+
+  /**
+   * Validate if the timer task is stopped
+   */
+  public abstract boolean isStopped();
```

Comment:
Would this break a class which extends HelixTimerTask in an external package?

Reply:
If this is for test only, please keep it in the implementation classes and keep it protected.

Reply:
The only exposed interface from ZkHelixManager is `List<HelixTimerTask> _controllerTimerTasks = xxx` and the actual implementation class is `static class StatusDumpTask` which is not exposed neither. How could I create a protected method and make it only visible for the test?

Reply:
One of the options is to cast the HelixTimerTask to a wrapper subclass only visible in test folders. And it assumes the objects in the list will be of type `StatusDumpTask`